Add a file path section in the first notebook! (#231)
AlexsLemonade/training-modules: intro-to-R-tidyverse/01-intro_to_base_R-live.Rmd

levels(iris$Species)
```

> ### Files and directories

Reviewer: Looking at this in the context of the new exercises pull request...

Should this be H2? It seems to be somewhat separate from the section on data frames.
```suggestion
## Files and directories
```

> In most situations, we will be reading in data from a file and using it as a data frame. 

Reviewer 2: ```suggestion
In many situations, we will be reading in tabular data from a file and using it as a data frame. 
```

To practice, we will read in a file we will be using in the next notebook as well, `gene_results_GSE44971.tsv`, in the `data` folder. 

> File paths are relative to where this notebook file (.Rmd) is saved.

Reviewer 2: ```suggestion
File paths are relative to the location where this notebook file (.Rmd) is saved.
```

Here we will use a function, `read_tsv()` from the `readr` package.
Before we are able to use the function, we have to load the package using `library()`. 

```{r tidyverse}
library(readr)
```

> `file.path()` adds a file path separator (`/` on Mac and Linux operating system, which is the operating system that our RStudio Server runs on) between separate folders or directory.

Reviewer 2: ```suggestion
`file.path()` creates a properly formatted file path by adding a path separator (`/` on Mac and Linux operating system, which is the operating system that our RStudio Server runs on) between separate folders or directories.
```

Because file path separators can differ between your computer and the computer of someone who wants to use your code, we use `file.path()` instead of typing out `"data/gene_results_GSE44971.tsv"`.
Each _argument_ to `file.path()` is a directory or file name.

> You'll notice each argument is in quotes, we specify `data` first because it the file, `gene_results_GSE44971.tsv` is in the `data` folder. 

Reviewer 2: ```suggestion
You'll notice each argument is in quotes, we specify `data` first because the file, `gene_results_GSE44971.tsv` is in the `data` folder. 
```

```{r file.path}

> file.path("data", "gene_results_GSE44971.tsv")

Reviewer: Would suggest including a step where you store the output of `file.path()` to a variable

```

> The resulting data frame will be stored in a variable named `stats_df`.

Reviewer 2: I agree with the previous comment that we should store the file path as a variable before using it below (nesting is potentially confusing at this stage).

```suggestion
Now we are ready to use `read_tsv()` to read the file into R.
The resulting data frame will be stored in a variable named `stats_df`.
```
